modules/andes-core/broker/src/main/java/org/wso2/andes/server/binding/BindingFactory.java

```diff
@@ -182,10 +182,15 @@ private synchronized boolean makeBinding(String bindingKey, AMQQueue queue, Exch
 
             queue.addQueueDeleteTask(binding);
             exchange.addCloseTask(binding);
-            queue.addBinding(binding);
-            exchange.addBinding(binding);
-            getConfigStore().addConfiguredObject(binding);
-            binding.logCreation();
+            try {
+                queue.addBinding(binding);
+                exchange.addBinding(binding);
+                getConfigStore().addConfiguredObject(binding);
+                binding.logCreation();
+            } catch (AndesException e) {
+                _bindings.remove(binding);
+                throw new AMQInternalException("Not permitted. Binding Already exists for this queue", e);
+            }
```

Comment:
Can we throw a generic message here something like 'Adding binding failed'. 'e' will contain the real cause. Otherwise if any other exception occur here we assume everything as not permitted.